src/rules.js

```diff
@@ -198,7 +198,7 @@ inline.tag = edit(inline.tag)
   .replace('attribute', inline._attribute)
   .getRegex();
 
-inline._label = /(?:\[[^\[\]]*\]|\\.|`[^`]*`|[^\[\]\\`])*?/;
+inline._label = /(?:\[(?:\\.|[^\[\]])*\]|\\.|`[^`]*`|[^\[\]\\`])*?/;
```

Comment:
@davisjam is this regex safe from ReDOS?

Reply:
Sadly, it has moved from linear to exponential.

[Visualization](https://regexper.com/#%2F%28%3F%3A%5C%5B%28%3F%3A%5C%5C.%7C%5B%5E%5C%5B%5C%5D%5D%29*%5C%5D%7C%5C%5C.%7C%60%5B%5E%60%5D*%60%7C%5B%5E%5C%5B%5C%5D%5C%5C%60%5D%29*%3F%2F) -- The problem is in the top group. The OR can be satisfied on the top or the bottom using `\\a`, and if the subsequent `]` is missing then you get exponential backtracking.

Example misbehavior:
```
/\[(?:\\.|[^\[\]])*\]/.exec("[" + "\\a".repeat(300));
```

Reply:
what about removing `\\` from the second path (`/\[(?:\\.|[^\\\[\]])*\]/`)? that way the pnly path it can take is determined by whether it sees a `\\`

Reply:
@davisjam thanks for your help on this. Your regex experience is invaluable.